src/rgw/rgw_http_client.h

```diff
@@ -102,6 +102,52 @@ class RGWHTTPClient
   int get_req_retcode();
 };
 
+
+class RGWHTTPHeadersCollector : public RGWHTTPClient {
+public:
+  /* Case insensitive comparator for containers carrying HTTP headers. */
+  struct CILess : public std::binary_function<std::string, std::string, bool> {
+    bool operator()(const std::string& lhs,
+                    const std::string& rhs) const {
+      return ::strcasecmp(lhs.c_str(), rhs.c_str()) < 0 ;
```

Comment:
I find calling out to strcasecmp odd here.  Could we use boost::iequals (boost/algorithm/string) here?